### PR TITLE
feat: revote

### LINKED ISSUE #2208: OpenGov. Revote. Revote screen
OpenGov. Revote. Revote screen

- [x] show About voting popup
- [x] calculate available amount of tokens for voting
- [x] enter the amount of tokens for voting
- [x] enter the conviction (0.1x, 1x, etc)
- [x] show how Governance lock changes
- [x] show locking period
- [x] select Aye/Nay/Abstain
- [x] for Abstain voting show multiplier 0.1x - [mockup](https://www.figma.com/design/ylvl7Ox25UuZj73Rn85DK7/OpenGov-voting?node-id=685-46346&t=NP1elzrHA6b2XQZF-4)
- [x] show network fee

- [x] Validations ([doc](https://docs.google.com/document/d/1AWNpTQbmfgLMrILjx1n8UClwPattQ1-6mA3n9Geosms/edit#heading=h.gwu7jx1ftazg))
- [x] Extrinsics ([doc](https://docs.google.com/document/d/1AWNpTQbmfgLMrILjx1n8UClwPattQ1-6mA3n9Geosms/edit#heading=h.uni5bokpvdu5))

Wallets:
- [x] Single Polkadot Vault wallet
- [x] Proxied wallet
- [x] WalletConnect, Nova Wallet wallet
- [x] Multishard wallet, Polkadot Vault wallet
- [x] Multisig wallet

[Figma mockups](https://www.figma.com/design/ylvl7Ox25UuZj73Rn85DK7/OpenGov-voting?node-id=655-72380&t=NP1elzrHA6b2XQZF-4)

### PR DESCRIPTION
**Pull request by johnthecat**
Closes #2208

### CODE REVIEW

**Comment by pgolovkin** [comment]
Please, don't merge it until the next release.

File: src/renderer/entities/governance/ui/VoteCharts/VoteChart/VoteChart.tsx [hunk 1]
```diff
@@ -28,7 +28,10 @@ export const VoteChart = ({ aye, nay, pass, descriptionPosition = 'tooltip', bgC
         </>
       )}
       <div
-        className="vote-chart-point"
+        className={cnTw(
+          'absolute flex h-4 w-1.5 translate-x-[-50%] justify-center bg-border-dark',
+          'after:contest-[""] after:block after:h-full after:w-0.5 after:rounded-sm after:bg-border-dark',
```
Comment (Asmadek) [inline src/renderer/entities/governance/ui/VoteCharts/VoteChart/VoteChart.tsx]: Is it typo?  Should be `after:content-[""]`?

File: src/renderer/entities/transaction/lib/transactionBuilder.ts [hunk 2]
```diff
@@ -412,6 +413,27 @@ function buildVote({ chain, accountId, referendumId, trackId, vote }: VoteParams
...
+  vote: TransactionVote;
+};
+
+function buildRevote({ chain, accountId, referendumId, trackId, vote }: RevoteParams): RevoteTransaction {
+  return {
+    chainId: chain.chainId,
+    address: toAddress(accountId, { prefix: chain.addressPrefix }),
+    type: TransactionType.REVOTE,
+    args: {
+      track: trackId,
```
Comment (Asmadek) [inline src/renderer/entities/transaction/lib/transactionBuilder.ts:430]: Do we really need track here? You remove `removeVote` which use track from extrinsic for revote

File: src/renderer/pages/Governance/ui/Governance.tsx [hunk 3]
```diff
@@ -149,17 +161,31 @@ export const Governance = () => {
           />
         )}
 
+      {showRevoteModal &&
+        nonNullable(network) &&
```
Comment (Asmadek) [inline src/renderer/pages/Governance/ui/Governance.tsx:165]: What is your opinion maybe add possibility to provide several params for nonNullable?

**Review by Asmadek — APPROVED** [review]

**Review by sokolova-an — APPROVED** [review]

File: src/renderer/pages/Governance/ui/Governance.tsx (hunk 3 above)
Comment (johnthecat) [reply src/renderer/pages/Governance/ui/Governance.tsx:165]: I've tried to write similar helpers, but typescript type narrowing don't work in such cases
Idea was smth like

```ts
type NonNullableTuple<Tuple extends [...any[]]> = {
  [Index in keyof Tuple]: NonNullable<Tuple[Index]>;
} & { length: Tuple['length'] };

function nonNullables<const T extends unknown[]>(values: T): values is NonNullableTuple<T> {
  return values.every((value) => value !== null && value !== undefined);
}
```

**Pull request merged**